src/Redux.js

```diff
@@ -26,6 +29,15 @@ export default class Redux {
     return this.dispatchFn(action);
   }
 
+  middleware(next) {
+    const recurse = (action) =>
+      typeof action === 'function' ?
+        action(recurse, ::this.getState) :
+        next(action);
+
+    return recurse;
+  }
+
```

Comment:
Hmm, on second thought this really should be its own module, implemented using higher-order middleware. Otherwise there's no way to pass middleware to `createDispatcher()` without overriding it.